### PR TITLE
Lnk 2116 logging improvements

### PR DESCRIPTION
**Pull request by edward-miller-lcg**

### CODE REVIEW

**Review by seanmcilvenna — APPROVED** [review]
Generally looks good. Would like to know if the reportId gets populated into `logging_event_property` table (or whatever it's called)... Ultimately would like to be able to perform the following select, if at all possible:
`SELECT * FROM logging_event e JOIN logging_event_property p ... WHERE p.name = 'reportId' and p.value = '12345'`

File: api/src/main/java/com/lantanagroup/link/api/controller/ReportController.java [hunk 1]
```diff
@@ -460,6 +463,8 @@ private Report generateResponse(TenantService tenantService, LinkCredentials use
     logger.info("Statistics for report {} are:\n{}", report.getId(), this.stopwatchManager.getStatistics());
     this.stopwatchManager.reset();
 
+    MDC.clear();
```
Comment (smailliwcs) [inline api/src/main/java/com/lantanagroup/link/api/controller/ReportController.java:466]: Need to put this in a `finally`, where everything below the `MDC.put` above forms the body of the corresponding `try`.

**Review by smailliwcs — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/lantanagroup/link/api/ReportGenerator.java [hunk 2]
```diff
@@ -66,8 +68,12 @@ public void generate(QueryPhase queryPhase) throws ExecutionException, Interrupt
     AtomicInteger progress = new AtomicInteger(0);
     List<PatientOfInterestModel> pois = measureContext.getPatientsOfInterest(queryPhase);
 
+    var contextMapCopy = MDC.getCopyOfContextMap();
+
+    forkJoinPool.submit(() -> pois.parallelStream().forEach(patient -> {
+
+                MDC.setContextMap(contextMapCopy);
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply api/src/main/java/com/lantanagroup/link/api/ReportGenerator.java:75]: Is it that you want to tag the report Id there?

File: api/src/main/java/com/lantanagroup/link/api/controller/ReportController.java [hunk 3]
```diff
@@ -408,31 +411,31 @@ private Report generateResponse(TenantService tenantService, LinkCredentials use
 
       // Scoop the data for the patients and store it
       if (config.isSkipQuery() || skipQuery) {
-        logger.info("Skipping initial query and store");
+        logger.info("Skipping initial query and store", report.getId());
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply api/src/main/java/com/lantanagroup/link/api/controller/ReportController.java:414]: ah, I meant to remove this. I was testing the logback funtionality and wanted to see how DBAppender reacted to an arg with no placeholder in the string. I'll work to remove it.

File: api/src/main/resources/logback.xml [hunk 4]
```diff
@@ -5,6 +5,6 @@
     <property name="LOGBACK_ROLLINGPOLICY_MAX_HISTORY" value="0"/>
     <include resource="org/springframework/boot/logging/logback/base.xml"/>
 
-    <logger name="com.lantanagroup.link" level="DEBUG"/>
+    <logger name="com.lantanagroup.link" level="INFO" />
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply api/src/main/resources/logback.xml:8]: I can roll back this change.

**Review by smailliwcs — APPROVED** [review]

**Pull request merged**